### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by abizer14**

### CODE REVIEW

File: tests/SingleTest.robot [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+*** Variables ***
+${WEBSITE_URL}=     https://bstackdemo.com
+
+*** Test Cases ***
+Bstack Website Login
+    Add Implicit Wait
+    Click on Sign In
+    Enter Credentials   demouser    testingisfun99
+    Click on Login
+    Verify username     demouser
```
Comment (Ankit098) [inline tests/SingleTest.robot]: This test case seems to be breaking for firefox..

File: config/KeywordsFile.robot [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    ${retrived_username}=   get text    css=.username
+    run keyword if    "${retrived_username}" == "${assert_username}"    mark test status    passed    Username validated!
+    run keyword if    "${retrived_username}" != "${assert_username}"    mark test status    failed    Username validation failed!
+    element should contain    css=.username    ${assert_username}
+
+Verify site content
+    ${site_content}=   get text    css=body
+    run keyword if    "${site_content}" == "Up and running"    mark test status    passed    Site Content validated!
+    run keyword if    "${site_content}" != "Up and running"    mark test status    failed    Site Content validation failed!
+    element should contain    css=body    Up and running
```
Comment (Ankit098) [inline config/KeywordsFile.robot]: EoF fixes, here and in other few files as well..

File: config/SingleCaps.robot [hunk 3]
```diff
@@ -0,0 +1,2 @@
+*** Variables ***
+&{SINGLE_CAPS}      name=single test    build=browserstack-robot-framework    browser=Edge    browser_version=latest    os=Windows    os_version=10
```
Comment (Ankit098) [inline config/SingleCaps.robot]: For all sample repos we are keeping the build name as `browserstack-build-1`, let's update this for local and parallel as well.

File: config/KeywordsFile.robot [hunk 4]
```diff
@@ -0,0 +1,56 @@
+*** Settings ***
+Library    SeleniumLibrary
+Library    mark-test-status.py
+Library    manage-local-testing.py
+
+*** Variables ***
+${USERNAME}=     %{BROWSERSTACK_USERNAME}
+${ACCESSKEY}=    %{BROWSERSTACK_ACCESS_KEY}
+${REMOTE_URL}=  https://${USERNAME}:${ACCESSKEY}@hub-cloud.browserstack.com/wd/hub
```
Comment (Ankit098) [inline config/KeywordsFile.robot]:
```suggestion
${REMOTE_URL}=  https://${USERNAME}:${ACCESSKEY}@hub.browserstack.com/wd/hub
```

File: tests/SingleTest.robot (hunk 1 above)
Comment (abizer14) [reply tests/SingleTest.robot]: Fixed under commit ecc30fbcb8f46a91f46ee21e8262c59793ed4e69

File: config/SingleCaps.robot (hunk 3 above)
Comment (abizer14) [reply config/SingleCaps.robot]: done under commit da231388133e9a65e54023ae9d56dbefaf500daf

File: config/mark-test-status.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+from robot.libraries.BuiltIn import BuiltIn
+from robot.libraries.BuiltIn import _Misc
+import robot.api.logger as logger
+from robot.api.deco import keyword
+
+@keyword("TEST STATUS")
+def testStatus(status, reason):
+     driver = BuiltIn().get_library_instance('SeleniumLibrary').driver
+     driver.execute_script('browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"'+status+'", "reason": "'+reason+'"}}')
+
```
Comment (Ankit098) [inline config/mark-test-status.py]: Please check if adding extra spaces/tabs at the last line, can remove that.

**Review by Ankit098 — CHANGES_REQUESTED** [review]
Minor comment about EoF changes.
Also we don't need to have the "single" test -> we only need to have the local and the parallel runs, need to update readme accordingly as well.

File: config/mark-test-status.py (hunk 5 above)
Comment (abizer14) [reply config/mark-test-status.py]: Done under commit a2da1fdd2acf4b06f3fcf97b1840e0fea5ecb9c3

**Review by Ankit098 — DISMISSED** [review]

**Review by alliv8 — COMMENTED** [review]
Let's ensure the test name and workflow is using Bstack demo similar to webdriverio, jest, testng etc

buildName: `browserstack-build-1`
eventual sessionName 
"BStack Demo - can add product to cart"


We check for they keyword "BStack" in session name to differentiate if users are running sample test or their own test

File: config/Config.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+{
+    "os" : "OS X",
+    "os_version" : "Big Sur",
+    "browser" : "Safari",
+    "browser_version" : "latest"
+},
+{
+    "device" : "Samsung Galaxy S22",
+    "os_version" : "12",
+    "real_mobile" : "true"
```
Comment (Ankit098) [inline config/Config.py]:
```suggestion
    "os_version" : "12"
```

Comment (Ankit098) [reply config/Config.py]: real_mobile is true (also false is not honoured) by default so we can remove this

**Review by Ankit098 — COMMENTED** [review]
minor comment, rest LGTM!

**Review by Ankit098 — APPROVED** [review]

**Review by francisf — APPROVED** [review]

**Pull request merged**